### PR TITLE
v1.14 Backports 2024-01-31

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #27207 (@pixiono)
 * [x] #29317 (@YutaroHayakawa)
 * [x] #30282 (@giorio94)
 * [x] #30167 (@viktor-kurchenko)
 * [x] #30403 (@f1ko)
 * [x] #30052 (@yingnanzhang666 :warning: minor conflict)
 * [x] #30410 (@julianwiedmann)
 * [x] #30399 (@tlcowling)
 * [x] #30219 (@dependabot[bot])
 * [x] #30236 (@soggiest)
 * [x] #30100 (@chaunceyjiang :warning: fixed conflict in tests)
 * [x] #30443 (@brlbil :warning: some conflicts, bumped upload-artifacts missed by commit)
 * [x] #30422 (@geakstr)
 * [x] #30503 (@qmonnet)
 * [x] #30416 (@joamaki :warning: redid renaming)
 * [x] #30329 (@3u13r)
 * [x] #30247 (@marseel)

PRs skipped due to conflicts:

 * #30154 (@ldelossa)
 * #30504 (@marseel)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 27207 29317 30282 30167 30403 30052 30410 30399 30219 30236 30100 30443 30422 30503 30416 30329 30247
```

### CODE REVIEW

**Review by viktor-kurchenko — APPROVED** [review]
LGTM, Thanks!

**Review by brlbil — APPROVED** [review]
Thanks!

**Review by chaunceyjiang — APPROVED** [review]
Thanks.

File: pkg/health/server/server.go [hunk 1]
```diff
@@ -336,12 +336,25 @@ func (s *Server) runActiveServices() error {
...
+			prober.updateIcmpStatus()
+			// (2) setNodes implementation doesn't override results for existing nodes.
+			// (4) Remove stale nodes so we don't report them in metrics before updating results
 			prober.setNodes(nodesAdded, nodesRemoved)
+			// (4) Update results without stale nodes
+			s.updateCluster(prober.getResults())
+			// (3) Cleanup icmp results for next iteration of probing
+			prober.clearIcmpStatus()
 		}
 		s.updateCluster(prober.getResults())
```
Comment (marseel) [inline pkg/health/server/server.go]: This line needs to be deleted. Maybe we should consider though backporting this change too https://github.com/cilium/cilium/pull/29745 instead so we don't have conflicts here.

**Review by marseel — CHANGES_REQUESTED** [review]

**Review by giorio94 — APPROVED** [review]
My commit looks good, thanks!

**Review by qmonnet — APPROVED** [review]

File: pkg/health/server/server.go (hunk 1 above)
Comment (joamaki) [reply pkg/health/server/server.go]: I've dropped the backport of the PR for now.

**Comment by joamaki** [comment]
/test-backport-1.14

**Review by marseel — APPROVED** [review]

**Review by geakstr — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]

**Review by YutaroHayakawa — APPROVED** [review]
Mine looks good. Thanks!

**Review by learnitall — APPROVED** [review]

**Comment by joamaki** [comment]
@cilium/ci-structure @cilium/github-sec could someone from these review?

**Review by pchaigno — APPROVED** [review]

**Pull request merged**